File: web-sources/S8-pkgs-ui-carbide/boxes/b2/B2ColumnsStruct.js

import { S8WebFront } from "/S8-pkgs-ui-carbide/S8WebFront.js";
import { S8Object } from "/S8-api/S8Object.js";
import { B2Column } from "./B2Column.js";



S8WebFront.CSS_import('/S8-pkgs-ui-carbide/boxes/b2/B2Box.css');


/**
 * 
 */
export class B2ColumnsStruct extends S8Object {

    /** @type {HTMLDivElement} */
    wrapperNode;


    /** */
    constructor() {
        super();
        this.wrapperNode = document.createElement("div");
        this.wrapperNode.classList.add("b2struct-columns");
    }


    /** @returns */
    getEnvelope() {
        return this.wrapperNode;
    }


    /** @param {B2Column[]} columns */
    S8_set_columns(columns) {
        S8WebFront.setS8Elements(this.wrapperNode, columns);
    }


    /** render */
    S8_render(){}


    /** dispose */
    S8_dispose(){}
}

File: sources/com/s8/pkgs/ui/carbide/boxes/b2/B2ColumnsStruct.java
package com.s8.pkgs.ui.carbide.boxes.b2;

import com.s8.api.web.S8WebFront;
import com.s8.pkgs.ui.carbide.HTML_S8WebNode;


/**
 * 
 * @author pierreconvert
 *
 */
public class B2ColumnsStruct extends HTML_S8WebNode {
	
	
	
	public final static String JS_TYPENAME = "/B2ColumnsStruct";
	
	

	
	/**
	 * 
	 * @param front
	 */
	
	public B2ColumnsStruct(S8WebFront front) {
		super(front, B2Box.ROOT_WEBPATH + JS_TYPENAME);
	}
	
	


	/**
	 * Columns
	 * @param size
	 */
	public void setColumns(B2Column... columns) {
		vertex.outbound().setObjectListField("columns", columns);
	}
	
	
}
